assembly/src/assembler/instruction/mod.rs

```diff
@@ -235,7 +235,15 @@ impl Assembler {
             Instruction::CDrop => span.add_ops([CSwap, Drop]),
             Instruction::CDropW => span.add_ops([CSwapW, Drop, Drop, Drop, Drop]),
 
-            Instruction::PushConstants(imms) => env_ops::push(imms, span),
+            Instruction::PushU8(imms) => env_ops::push(&[Felt::from(*imms)], span),
+            Instruction::PushU16(imms) => env_ops::push(&[Felt::from(*imms)], span),
+            Instruction::PushU32(imms) => env_ops::push(&[Felt::from(*imms)], span),
+            Instruction::PushFelt(imms) => env_ops::push(&[*imms], span),
+            Instruction::PushWord(imms) => env_ops::push(imms, span),
+            Instruction::PushU8List(imms) => cast_push_vec(imms, span),
+            Instruction::PushU16List(imms) => cast_push_vec(imms, span),
+            Instruction::PushU32List(imms) => cast_push_vec(imms, span),
+            Instruction::PushFeltList(imms) => env_ops::push(imms, span),
```

Comment:
Instead of trying to use the same `env_ops::push()` function everywhere, I'd create two functions there: `push_one()` and `push_many()`. These functions could look something like this:
```Rust
pub fn push_one<T>(imm: T, span: &mut SpanBuilder) -> Result<Option<CodeBlock>, AssemblyError>
where
    T: Into<Felt>,
{
    push_felt(span, imm.into());
    Ok(None)
}

pub fn push_many<T>(imms: &[T], span: &mut SpanBuilder) -> Result<Option<CodeBlock>, AssemblyError>
where
    T: Into<Felt> + Copy,
{
    imms.iter().for_each(|imm| push_felt(span, (*imm).into()));
    Ok(None)
}
```